be/src/vec/exec/format/parquet/vparquet_column_chunk_reader.cpp

```diff
@@ -66,44 +68,48 @@ Status ColumnChunkReader::init() {
     RETURN_IF_ERROR(get_block_compression_codec(_metadata.codec, &_block_compress_codec));
     if (_metadata.__isset.dictionary_page_offset) {
         // seek to the directory page
-        _page_reader->seek_to_page(_metadata.dictionary_page_offset);
-        // Parse dictionary data when reading
-        // RETURN_IF_ERROR(_page_reader->next_page_header());
-        // RETURN_IF_ERROR(_decode_dict_page());
+        _page_reader->seek_page(_metadata.dictionary_page_offset);
     } else {
         // seek to the first data page
-        _page_reader->seek_to_page(_metadata.data_page_offset);
+        _page_reader->seek_page(_metadata.data_page_offset);
     }
     _state = INITIALIZED;
+
+    // the first page maybe directory page even if _metadata.__isset.dictionary_page_offset == false
+    RETURN_IF_ERROR(_page_reader->parse_page_header());
+    if (_page_reader->get_page_header()->type == tparquet::PageType::DICTIONARY_PAGE) {
+        // Parse dictionary data when reading
+        RETURN_IF_ERROR(_decode_dict_page());
+    }
     return Status::OK();
 }
 
 Status ColumnChunkReader::next_page() {
-    if (_state == HEADER_PARSED) {
-        return Status::OK();
-    }
     if (UNLIKELY(_state == NOT_INIT)) {
         return Status::Corruption("Should initialize chunk reader");
     }
     if (UNLIKELY(_remaining_num_values != 0)) {
         return Status::Corruption("Should skip current page");
     }
-    RETURN_IF_ERROR(_page_reader->next_page_header());
-    if (_page_reader->get_page_header()->type == tparquet::PageType::DICTIONARY_PAGE) {
-        // the first page maybe directory page even if _metadata.__isset.dictionary_page_offset == false,
-        // so we should parse the directory page in next_page()
-        RETURN_IF_ERROR(_decode_dict_page());
-        // parse the real first data page
-        return next_page();
-    } else if (_page_reader->get_page_header()->type == tparquet::PageType::DATA_PAGE_V2) {
+
+    if (_offset_index) {
```

Comment:
Don't add is-else logic, the underlying `PageReader` supply the unified interface.